src/Collapsible.js

```diff
@@ -1,93 +1,79 @@
-import React, { Component } from 'react';
+import React, { Children, useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 
-class Collapsible extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      activeKey: props.defaultActiveKey
-    };
-
-    this.renderItem = this.renderItem.bind(this);
-    this.handleSelect = this.handleSelect.bind(this);
-  }
-
-  componentDidMount() {
-    if (typeof M !== 'undefined') {
-      this.instance = M.Collapsible.init(this._collapsible, {
-        accordion: this.props.accordion,
-        ...this.props.options
-      });
-    }
-  }
-
-  componentWillUnmount() {
-    if (this.instance) {
-      this.instance.destroy();
-    }
-  }
-
-  render() {
-    const { accordion, popout, className, children, ...props } = this.props;
-
-    delete props.defaultActiveKey;
-
-    const collapsible = accordion ? 'accordion' : 'expandable';
-    const classes = {
-      collapsible,
-      expandable: accordion !== true,
-      popout
-    };
-
-    return (
-      <ul
-        ref={node => {
-          this._collapsible = node;
-        }}
-        className={cx(className, classes)}
-        data-collapsible={collapsible}
-        {...props}
-      >
-        {React.Children.map(children, this.renderItem)}
-      </ul>
-    );
-  }
-
-  renderItem(child, key) {
+const Collapsible = ({
+  children,
+  className,
+  accordion,
+  options,
+  defaultActiveKey,
+  popout,
+  onSelect,
+  ...props
+}) => {
+  const _collapsible = useRef(null);
+  const collapsible = accordion ? 'accordion' : 'expandable';
+  const [activeKey, setActiveKey] = useState(defaultActiveKey);
+
+  const renderItem = (child, key) => {
```

Comment:
Ok, after reading the article you so kindly shared, I think this `renderItem` should be wrapped in a memoized helper (your choice) :)